src/Block/Service/AbstractCategoriesBlockService.php

```diff
@@ -44,14 +44,33 @@ abstract class AbstractCategoriesBlockService extends AbstractClassificationBloc
     private $categoryAdmin;
 
     /**
-     * @param string $name
+     * @param string|Environment                               $twigOrDeprecatedName
+     * @param EngineInterface|ContextManagerInterface          $contextManagerOrDeprecatedTemplating
+     * @param ContextManagerInterface|CategoryManagerInterface $categoryManagerOrDeprecatedContextManager
+     * @param CategoryManagerInterface|AdminInterface          $categoryAdminOrDeprecatedCategoryManager
+     * @param AdminInterface|null                              $deprecatedCategoryAdmin
      */
-    public function __construct($name, EngineInterface $templating, ContextManagerInterface $contextManager, CategoryManagerInterface $categoryManager, AdminInterface $categoryAdmin)
+    public function __construct($twigOrDeprecatedName, $contextManagerOrDeprecatedTemplating, $categoryManagerOrDeprecatedContextManager, $categoryAdminOrDeprecatedCategoryManager, $deprecatedCategoryAdmin)
     {
-        parent::__construct($name, $templating, $contextManager);
-
-        $this->categoryManager = $categoryManager;
-        $this->categoryAdmin = $categoryAdmin;
+        // NEXT_MAJOR: remove the if block
+        if (\is_string($twigOrDeprecatedName)) {
+            parent::__construct(
```

Comment:
We don't need any deprecation warning here, because the parent class will trigger one.